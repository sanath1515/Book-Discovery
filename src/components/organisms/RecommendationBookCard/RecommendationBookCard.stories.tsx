import React from "react";
import RecommendationBookCard from "./RecommendationBookCard";

export default {
  title: "organisms/RecommendationBookCard",
  component: RecommendationBookCard,
};

const Template = () => (
  <RecommendationBookCard
    title="Nature Chemistry"
    authorName="J D Lee"
    image="https://cdn.zeplin.io/5f03041d8c8fc11335546618/assets/920F4926-3BD5-47FB-A32A-32C03DE1BB71.png"
    id={"1"}
  />
);

export const Default = Template.bind({});
