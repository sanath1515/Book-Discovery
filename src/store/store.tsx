import { createStore } from 'redux';
import { booksReducer } from './reducers';

const store = createStore(booksReducer);

export default store;