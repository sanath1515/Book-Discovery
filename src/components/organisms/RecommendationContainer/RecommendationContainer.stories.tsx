import React from 'react';
import { RecommendationBookCardProps } from '../RecommendationBookCard/RecommendationBookCard';
import RecommendationContainer from './RecommendationContainer';

const bookData: RecommendationBookCardProps[] = [
    {
        "id":"1",
        "title": "Crystal chemistry",
        "authorName": "Joe Pearson",
        "image": "https://cdn.zeplin.io/5f03041d8c8fc11335546618/assets/E3EEB4FA-4DF1-491D-8A51-DC1C1921FB60.png",
        
    },
    {
        "id":"2",
        "title": "Inorganic chemistry",
        "authorName": "Joe Pearson",
        "image": "https://cdn.zeplin.io/5f03041d8c8fc11335546618/assets/03991C69-3988-432F-AD72-535D7FCFFC54.png",
    
    },
    {
        "id":"3",
        "title": "Bioorganic chemistry",
        "authorName": "Joe Pearson",
        "image": "https://cdn.zeplin.io/5f03041d8c8fc11335546618/assets/19118712-5417-4364-9FC6-59FA289CE148.png",
    },
    {
        "id":"4",
        "title": "Basic Physics",
        "authorName": "Joe Pearson",
        "image": "https://cdn.zeplin.io/5f03041d8c8fc11335546618/assets/BFB91888-5030-4E35-878F-68EC357D95CE.png",
    },
  ]

export default {
    title: 'organisms/RecommendationContainer',
    component: RecommendationContainer
};

const Template = (args) => <RecommendationContainer data={bookData} />;

export const Default = Template.bind({});
