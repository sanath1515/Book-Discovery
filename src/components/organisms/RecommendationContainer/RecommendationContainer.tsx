import React, { useState } from "react";
import RecommendationBookCard from "../RecommendationBookCard/RecommendationBookCard";
import { Grid, makeStyles, Typography } from "@material-ui/core";
import { COLORS } from "../../../theme/constants";
import ArrowForwardIosIcon from "@material-ui/icons/ArrowForwardIos";
import theme from "../../../theme/theme";
import ArrowBackIosIcon from "@material-ui/icons/ArrowBackIos";
import { bookProp } from "../../../data/type";

export type RecommendationContainerProps = {
  data: Array<bookProp>;
  className?: string;
  heading?: string;
  subHeading?: string;
  mainHeading?:string;
};
const useStyles = makeStyles({
  heading: {
    width: "475px",
    height: "38px",
    color: COLORS.CHARCOAL_GREY,
    float: "left",
    marginBottom: theme.spacing(0.5)
  },
  mainHeading: {
    width: "475px",
    height: "38px",
    color: COLORS.CHARCOAL_GREY_100,
    float: "left",
    marginBottom: theme.spacing(2.5)
  },
  headingLine:{
    width: "475px",
    height: "38px",
    color: COLORS.CHARCOAL_GREY,
    marginBottom: theme.spacing(1)
  },
  text: {
    width: "68px",
    height: "24px",
    margin: theme.spacing(1, 1, 4, 0),
    color: COLORS.PRIMARY_COLOR_500,
    cursor: "pointer",
    float: "right",
  },
  icon: {
    color: COLORS.PRIMARY_COLOR_500,
    margin: theme.spacing(2.5, 2, 4, 0),
    fontSize: "12px",
    cursor: "pointer",
    float: "right",
  },
  grid: {
    display: "flex",
    justifyContent: "space-between",
    width: "100%",
    maxWidth: "1110px",
  },
  subHeading: {
    color: COLORS.ICON_500,
    float: "left",
  },
});
const RecommendationContainer: React.FC<RecommendationContainerProps> = ({
  data,
  heading,
  subHeading,
  mainHeading
}) => {
  const classes = useStyles();
  const [range, setRange] = useState({ low: 0, high: 4 });
  const [text, setText] = useState("See more");
  const [iconMark, setIconMark] = useState(true);

  const handleOnClick = () => {
    console.log(range.low + " " + range.high);
    if (range.high == data.length) {
      let low = 0;
      let high = 4;
      setText("See More");
      setIconMark(true);
      setRange({ low, high });
    } else {
      let low = range.low + 1;
      let high = range.high + 1;
      setRange({ low, high });
      if (high == data.length) {
        setIconMark(false);
        setText("Go Back");
      }
    }
  };
  const recommendationCards = data
    .slice(range.low, range.high)
    .map((dataItem) => {
      return (
        <RecommendationBookCard
          id={dataItem.id!}
          authorName={dataItem.author!}
          title={dataItem.title!}
          image={dataItem.bookImage!}
        />
      );
    });
  const Heading = (
    <Grid>
      <Typography variant="h3" className={classes.headingLine}>
        {heading}
      </Typography>
    </Grid>
  );
  return (
    <div>
      {heading && subHeading && Heading}
      <Grid>
        {!subHeading && (
          <Typography variant="h3" className={mainHeading?classes.mainHeading:classes.heading}>
            {mainHeading?mainHeading:heading}
          </Typography>
        )}
        <Typography variant="h5" className={classes.subHeading}>
          {subHeading}
        </Typography>
        {iconMark && (
          <ArrowForwardIosIcon
            className={classes.icon}
            onClick={handleOnClick}
            data-testid="seemore"
          />
        )}

        {!iconMark && (
          <ArrowBackIosIcon className={classes.icon} onClick={handleOnClick} />
        )}
        <Typography
          variant="subtitle2"
          className={classes.text}
          onClick={handleOnClick}
          data-testid="seeless"
        >
          {text}
        </Typography>
      </Grid>
      <Grid className={classes.grid}>{recommendationCards}</Grid>
    </div>
  );
};

export default RecommendationContainer;
