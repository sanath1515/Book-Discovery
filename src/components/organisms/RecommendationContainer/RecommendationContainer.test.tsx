import { fireEvent, render } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import store from "../../../store/store";
import { RecommendationBookCardProps } from "../RecommendationBookCard/RecommendationBookCard";
import RecommendationContainer from "./RecommendationContainer";

const bookData: RecommendationBookCardProps[] = [
  {
    id: "1",
    title: "Crystal chemistry",
    authorName: "Joe Pearson",
    image:
      "https://cdn.zeplin.io/5f03041d8c8fc11335546618/assets/E3EEB4FA-4DF1-491D-8A51-DC1C1921FB60.png",
  },
  {
    id: "2",
    title: "Inorganic chemistry",
    authorName: "Joe Pearson",
    image:
      "https://cdn.zeplin.io/5f03041d8c8fc11335546618/assets/03991C69-3988-432F-AD72-535D7FCFFC54.png",
  },
  {
    id: "3",
    title: "Bioorganic chemistry",
    authorName: "Joe Pearson",
    image:
      "https://cdn.zeplin.io/5f03041d8c8fc11335546618/assets/19118712-5417-4364-9FC6-59FA289CE148.png",
  },
  {
    id: "4",
    title: "Basic Physics",
    authorName: "Joe Pearson",
    image:
      "https://cdn.zeplin.io/5f03041d8c8fc11335546618/assets/BFB91888-5030-4E35-878F-68EC357D95CE.png",
  },
];

describe("Should render container correctly", () => {
  const component = (
    <Router>
      <Provider store={store}>
        <RecommendationContainer data={bookData} />
      </Provider>
    </Router>
  );

  it("Match Snapshot", () => {
    const { container } = render(
      <Router>
        <Provider store={store}>
          <RecommendationContainer data={bookData} />
        </Provider>
      </Router>
    );
    expect(container).toMatchSnapshot();
  });

  test("Should render correctly", () => {
    const { container } = render(
      <Router>
        <Provider store={store}>
          <RecommendationContainer data={bookData} />
        </Provider>
      </Router>
    );
    expect(container).toBeInTheDocument();
  });
  it("Should click seemore correctly", () => {
    const { getByTestId } = render(
      <Router>
      <Provider store={store}>
        <RecommendationContainer data={bookData} />
      </Provider>
    </Router>
    );
    expect(fireEvent.click(getByTestId("seemore"))).toBeTruthy();
  });
  it("Should click seeless correctly", () => {
    const { getByTestId } = render(
      <Router>
      <Provider store={store}>
        <RecommendationContainer data={bookData} />
      </Provider>
    </Router>
    );
    expect(fireEvent.click(getByTestId("seeless"))).toBeTruthy();
  });
});
