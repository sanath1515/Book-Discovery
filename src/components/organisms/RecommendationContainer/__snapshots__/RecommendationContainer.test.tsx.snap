// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render container correctly Match Snapshot 1`] = `
<div>
  <div>
    <div
      class="MuiGrid-root"
    >
      <h3
        class="MuiTypography-root makeStyles-heading-1 MuiTypography-h3"
      />
      <h5
        class="MuiTypography-root makeStyles-subHeading-7 MuiTypography-h5"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root makeStyles-icon-5"
        data-testid="seemore"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M5.88 4.12L13.76 12l-7.88 7.88L8 22l10-10L8 2z"
        />
      </svg>
      <h6
        class="MuiTypography-root makeStyles-text-4 MuiTypography-subtitle2"
        data-testid="seeless"
      >
        See more
      </h6>
    </div>
    <div
      class="MuiGrid-root makeStyles-grid-6"
    >
      <div
        class="MuiPaper-root MuiCard-root makeStyles-root-8 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <td
          class="MuiTableCell-root makeStyles-cell-22"
        >
          <img
            class="MuiCardMedia-root makeStyles-image-9 MuiCardMedia-media MuiCardMedia-img"
            data-testid="card"
          />
        </td>
        <div
          class="MuiCardContent-root makeStyles-content-21"
        >
          <div
            class="MuiCardActions-root makeStyles-titleActions-16 MuiCardActions-spacing"
          >
            <h6
              class="MuiTypography-root makeStyles-title-10 MuiTypography-h6"
            >
              Crystal chemistry
            </h6>
          </div>
          <div
            class="MuiCardActions-root makeStyles-actions-15 MuiCardActions-spacing"
          >
            <p
              class="MuiTypography-root makeStyles-author-11 MuiTypography-body2"
            >
              Author
            </p>
            <h6
              class="MuiTypography-root makeStyles-authorName-12 MuiTypography-subtitle2"
            />
          </div>
          <div
            class="MuiCardActions-root makeStyles-actions-15 MuiCardActions-spacing"
          >
            <span
              aria-label="4 Stars"
              class="MuiRating-root MuiRating-sizeSmall MuiRating-readOnly"
              role="img"
            >
              <span>
                <span
                  class="MuiRating-icon MuiRating-iconFilled"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                    />
                  </svg>
                </span>
              </span>
              <span>
                <span
                  class="MuiRating-icon MuiRating-iconFilled"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                    />
                  </svg>
                </span>
              </span>
              <span>
                <span
                  class="MuiRating-icon MuiRating-iconFilled"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                    />
                  </svg>
                </span>
              </span>
              <span>
                <span
                  class="MuiRating-icon MuiRating-iconFilled"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                    />
                  </svg>
                </span>
              </span>
              <span>
                <span
                  class="MuiRating-icon MuiRating-iconEmpty"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                    />
                  </svg>
                </span>
              </span>
            </span>
            <p
              class="MuiTypography-root makeStyles-star-13 MuiTypography-body1"
            >
              4.0
            </p>
            <span
              class="MuiTypography-root makeStyles-rating-14 MuiTypography-overline"
            >
              (1,204 ratings)
            </span>
          </div>
        </div>
        <hr
          class="MuiDivider-root"
        />
        <div
          class="MuiCardActions-root makeStyles-actionIcons-23 MuiCardActions-spacing"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-bookmarkBorder-18"
            data-testid="bookmark"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M17 3H7c-1.1 0-1.99.9-1.99 2L5 21l7-3 7 3V5c0-1.1-.9-2-2-2zm0 15l-5-2.18L7 18V5h10v13z"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-reply-19"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 9V5l-7 7 7 7v-4.1c5 0 8.5 1.6 11 5.1-1-5-4-10-11-11z"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-playlist-20"
            data-testid="playlist"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M13 10H3c-.55 0-1 .45-1 1s.45 1 1 1h10c.55 0 1-.45 1-1s-.45-1-1-1zm0-4H3c-.55 0-1 .45-1 1s.45 1 1 1h10c.55 0 1-.45 1-1s-.45-1-1-1zm5 8v-3c0-.55-.45-1-1-1s-1 .45-1 1v3h-3c-.55 0-1 .45-1 1s.45 1 1 1h3v3c0 .55.45 1 1 1s1-.45 1-1v-3h3c.55 0 1-.45 1-1s-.45-1-1-1h-3zM3 16h6c.55 0 1-.45 1-1s-.45-1-1-1H3c-.55 0-1 .45-1 1s.45 1 1 1z"
            />
          </svg>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiCard-root makeStyles-root-8 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <td
          class="MuiTableCell-root makeStyles-cell-22"
        >
          <img
            class="MuiCardMedia-root makeStyles-image-9 MuiCardMedia-media MuiCardMedia-img"
            data-testid="card"
          />
        </td>
        <div
          class="MuiCardContent-root makeStyles-content-21"
        >
          <div
            class="MuiCardActions-root makeStyles-titleActions-16 MuiCardActions-spacing"
          >
            <h6
              class="MuiTypography-root makeStyles-title-10 MuiTypography-h6"
            >
              Inorganic chemistry
            </h6>
          </div>
          <div
            class="MuiCardActions-root makeStyles-actions-15 MuiCardActions-spacing"
          >
            <p
              class="MuiTypography-root makeStyles-author-11 MuiTypography-body2"
            >
              Author
            </p>
            <h6
              class="MuiTypography-root makeStyles-authorName-12 MuiTypography-subtitle2"
            />
          </div>
          <div
            class="MuiCardActions-root makeStyles-actions-15 MuiCardActions-spacing"
          >
            <span
              aria-label="4 Stars"
              class="MuiRating-root MuiRating-sizeSmall MuiRating-readOnly"
              role="img"
            >
              <span>
                <span
                  class="MuiRating-icon MuiRating-iconFilled"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                    />
                  </svg>
                </span>
              </span>
              <span>
                <span
                  class="MuiRating-icon MuiRating-iconFilled"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                    />
                  </svg>
                </span>
              </span>
              <span>
                <span
                  class="MuiRating-icon MuiRating-iconFilled"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                    />
                  </svg>
                </span>
              </span>
              <span>
                <span
                  class="MuiRating-icon MuiRating-iconFilled"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                    />
                  </svg>
                </span>
              </span>
              <span>
                <span
                  class="MuiRating-icon MuiRating-iconEmpty"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                    />
                  </svg>
                </span>
              </span>
            </span>
            <p
              class="MuiTypography-root makeStyles-star-13 MuiTypography-body1"
            >
              4.0
            </p>
            <span
              class="MuiTypography-root makeStyles-rating-14 MuiTypography-overline"
            >
              (1,204 ratings)
            </span>
          </div>
        </div>
        <hr
          class="MuiDivider-root"
        />
        <div
          class="MuiCardActions-root makeStyles-actionIcons-23 MuiCardActions-spacing"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-bookmarkBorder-18"
            data-testid="bookmark"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M17 3H7c-1.1 0-1.99.9-1.99 2L5 21l7-3 7 3V5c0-1.1-.9-2-2-2zm0 15l-5-2.18L7 18V5h10v13z"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-reply-19"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 9V5l-7 7 7 7v-4.1c5 0 8.5 1.6 11 5.1-1-5-4-10-11-11z"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-playlist-20"
            data-testid="playlist"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M13 10H3c-.55 0-1 .45-1 1s.45 1 1 1h10c.55 0 1-.45 1-1s-.45-1-1-1zm0-4H3c-.55 0-1 .45-1 1s.45 1 1 1h10c.55 0 1-.45 1-1s-.45-1-1-1zm5 8v-3c0-.55-.45-1-1-1s-1 .45-1 1v3h-3c-.55 0-1 .45-1 1s.45 1 1 1h3v3c0 .55.45 1 1 1s1-.45 1-1v-3h3c.55 0 1-.45 1-1s-.45-1-1-1h-3zM3 16h6c.55 0 1-.45 1-1s-.45-1-1-1H3c-.55 0-1 .45-1 1s.45 1 1 1z"
            />
          </svg>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiCard-root makeStyles-root-8 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <td
          class="MuiTableCell-root makeStyles-cell-22"
        >
          <img
            class="MuiCardMedia-root makeStyles-image-9 MuiCardMedia-media MuiCardMedia-img"
            data-testid="card"
          />
        </td>
        <div
          class="MuiCardContent-root makeStyles-content-21"
        >
          <div
            class="MuiCardActions-root makeStyles-titleActions-16 MuiCardActions-spacing"
          >
            <h6
              class="MuiTypography-root makeStyles-title-10 MuiTypography-h6"
            >
              Bioorganic chemistry
            </h6>
          </div>
          <div
            class="MuiCardActions-root makeStyles-actions-15 MuiCardActions-spacing"
          >
            <p
              class="MuiTypography-root makeStyles-author-11 MuiTypography-body2"
            >
              Author
            </p>
            <h6
              class="MuiTypography-root makeStyles-authorName-12 MuiTypography-subtitle2"
            />
          </div>
          <div
            class="MuiCardActions-root makeStyles-actions-15 MuiCardActions-spacing"
          >
            <span
              aria-label="4 Stars"
              class="MuiRating-root MuiRating-sizeSmall MuiRating-readOnly"
              role="img"
            >
              <span>
                <span
                  class="MuiRating-icon MuiRating-iconFilled"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                    />
                  </svg>
                </span>
              </span>
              <span>
                <span
                  class="MuiRating-icon MuiRating-iconFilled"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                    />
                  </svg>
                </span>
              </span>
              <span>
                <span
                  class="MuiRating-icon MuiRating-iconFilled"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                    />
                  </svg>
                </span>
              </span>
              <span>
                <span
                  class="MuiRating-icon MuiRating-iconFilled"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                    />
                  </svg>
                </span>
              </span>
              <span>
                <span
                  class="MuiRating-icon MuiRating-iconEmpty"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                    />
                  </svg>
                </span>
              </span>
            </span>
            <p
              class="MuiTypography-root makeStyles-star-13 MuiTypography-body1"
            >
              4.0
            </p>
            <span
              class="MuiTypography-root makeStyles-rating-14 MuiTypography-overline"
            >
              (1,204 ratings)
            </span>
          </div>
        </div>
        <hr
          class="MuiDivider-root"
        />
        <div
          class="MuiCardActions-root makeStyles-actionIcons-23 MuiCardActions-spacing"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-bookmarkBorder-18"
            data-testid="bookmark"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M17 3H7c-1.1 0-1.99.9-1.99 2L5 21l7-3 7 3V5c0-1.1-.9-2-2-2zm0 15l-5-2.18L7 18V5h10v13z"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-reply-19"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 9V5l-7 7 7 7v-4.1c5 0 8.5 1.6 11 5.1-1-5-4-10-11-11z"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-playlist-20"
            data-testid="playlist"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M13 10H3c-.55 0-1 .45-1 1s.45 1 1 1h10c.55 0 1-.45 1-1s-.45-1-1-1zm0-4H3c-.55 0-1 .45-1 1s.45 1 1 1h10c.55 0 1-.45 1-1s-.45-1-1-1zm5 8v-3c0-.55-.45-1-1-1s-1 .45-1 1v3h-3c-.55 0-1 .45-1 1s.45 1 1 1h3v3c0 .55.45 1 1 1s1-.45 1-1v-3h3c.55 0 1-.45 1-1s-.45-1-1-1h-3zM3 16h6c.55 0 1-.45 1-1s-.45-1-1-1H3c-.55 0-1 .45-1 1s.45 1 1 1z"
            />
          </svg>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiCard-root makeStyles-root-8 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <td
          class="MuiTableCell-root makeStyles-cell-22"
        >
          <img
            class="MuiCardMedia-root makeStyles-image-9 MuiCardMedia-media MuiCardMedia-img"
            data-testid="card"
          />
        </td>
        <div
          class="MuiCardContent-root makeStyles-content-21"
        >
          <div
            class="MuiCardActions-root makeStyles-titleActions-16 MuiCardActions-spacing"
          >
            <h6
              class="MuiTypography-root makeStyles-title-10 MuiTypography-h6"
            >
              Basic Physics
            </h6>
          </div>
          <div
            class="MuiCardActions-root makeStyles-actions-15 MuiCardActions-spacing"
          >
            <p
              class="MuiTypography-root makeStyles-author-11 MuiTypography-body2"
            >
              Author
            </p>
            <h6
              class="MuiTypography-root makeStyles-authorName-12 MuiTypography-subtitle2"
            />
          </div>
          <div
            class="MuiCardActions-root makeStyles-actions-15 MuiCardActions-spacing"
          >
            <span
              aria-label="4 Stars"
              class="MuiRating-root MuiRating-sizeSmall MuiRating-readOnly"
              role="img"
            >
              <span>
                <span
                  class="MuiRating-icon MuiRating-iconFilled"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                    />
                  </svg>
                </span>
              </span>
              <span>
                <span
                  class="MuiRating-icon MuiRating-iconFilled"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                    />
                  </svg>
                </span>
              </span>
              <span>
                <span
                  class="MuiRating-icon MuiRating-iconFilled"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                    />
                  </svg>
                </span>
              </span>
              <span>
                <span
                  class="MuiRating-icon MuiRating-iconFilled"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                    />
                  </svg>
                </span>
              </span>
              <span>
                <span
                  class="MuiRating-icon MuiRating-iconEmpty"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                    />
                  </svg>
                </span>
              </span>
            </span>
            <p
              class="MuiTypography-root makeStyles-star-13 MuiTypography-body1"
            >
              4.0
            </p>
            <span
              class="MuiTypography-root makeStyles-rating-14 MuiTypography-overline"
            >
              (1,204 ratings)
            </span>
          </div>
        </div>
        <hr
          class="MuiDivider-root"
        />
        <div
          class="MuiCardActions-root makeStyles-actionIcons-23 MuiCardActions-spacing"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-bookmarkBorder-18"
            data-testid="bookmark"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M17 3H7c-1.1 0-1.99.9-1.99 2L5 21l7-3 7 3V5c0-1.1-.9-2-2-2zm0 15l-5-2.18L7 18V5h10v13z"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-reply-19"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 9V5l-7 7 7 7v-4.1c5 0 8.5 1.6 11 5.1-1-5-4-10-11-11z"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-playlist-20"
            data-testid="playlist"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M13 10H3c-.55 0-1 .45-1 1s.45 1 1 1h10c.55 0 1-.45 1-1s-.45-1-1-1zm0-4H3c-.55 0-1 .45-1 1s.45 1 1 1h10c.55 0 1-.45 1-1s-.45-1-1-1zm5 8v-3c0-.55-.45-1-1-1s-1 .45-1 1v3h-3c-.55 0-1 .45-1 1s.45 1 1 1h3v3c0 .55.45 1 1 1s1-.45 1-1v-3h3c.55 0 1-.45 1-1s-.45-1-1-1h-3zM3 16h6c.55 0 1-.45 1-1s-.45-1-1-1H3c-.55 0-1 .45-1 1s.45 1 1 1z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
